src/main/java/org/mule/extension/db/internal/resolver/param/StoredProcedureParamTypeResolver.java

```diff
@@ -57,31 +57,42 @@ public StoredProcedureParamTypeResolver(DbTypeManager dbTypeManager) {
   @Override
   public Map<Integer, DbType> getParameterTypes(DbConnection connection, QueryTemplate queryTemplate, List<ParameterType> types)
       throws SQLException {
+    ResultSet procedureColumns = null;
     DatabaseMetaData dbMetaData = connection.getJdbcConnection().getMetaData();
 
+    String packageName = getStoreProcedureSchema(queryTemplate.getSqlText()).orElseGet(() -> "");
     String storedProcedureName = getStoredProcedureName(queryTemplate.getSqlText());
     if (dbMetaData.storesUpperCaseIdentifiers()) {
+      packageName = packageName.toUpperCase();
       storedProcedureName = storedProcedureName.toUpperCase();
     }
 
-    String storedProcedureSchema = getStoreProcedureSchema(queryTemplate.getSqlText()).orElse(null);
-    if (dbMetaData.storesUpperCaseIdentifiers() && storedProcedureSchema != null) {
-      storedProcedureSchema = storedProcedureSchema.toUpperCase();
-    }
+    try {
 
-    ResultSet procedureColumns =
-        dbMetaData.getProcedureColumns(connection.getJdbcConnection().getCatalog(), storedProcedureSchema, storedProcedureName,
-                                       "%");
+      procedureColumns =
+          dbMetaData.getProcedureColumns(packageName.isEmpty() ? connection.getJdbcConnection().getCatalog() : packageName,
+                                         connection.getJdbcConnection().getSchema(),
+                                         storedProcedureName, "%");
+      Map<Integer, DbType> paramTypes = getStoredProcedureParamTypes(connection, packageName, procedureColumns);
+
+
+      //if still unable to resolve, remove all catalog and schema filters
+      //and use only sp name and column pattern.
+      if (paramTypes.isEmpty()) {
```

Comment:
We should add some logging here. It might take a long time to the DB to response to this request. For example for Oracle 11g and driver ojdbc7 can take a really long time.